chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomService.kt

```diff
@@ -180,7 +182,17 @@ internal class RoomService(
                 messageLimit = messageLimit,
                 roomId = roomId,
                 cursorService = cursorsService,
-                membershipConsumer = { roomStore.applyMembershipEvent(roomId, it).map(::enrichEvent).forEach(emit) },
+                membershipConsumer = {
+                    if (it is MembershipSubscriptionEvent.InitialState
+                            && !roomsPassedInitialState.contains(roomId)) {
+                        //if it's the first initial state event we don't want to emit the callbacks for
+                        //people who have entered the room
```

Comment:
I would describe these users are those "who were already in the room", to maximise clarity, "who have entered" does suggest to me they did it recently.

Reply:
Good point! I've updated the comment for this!